### DISCUSSION TITLE
[Feature proposal] Add `-webkit-box` display type

### DISCUSSION THREAD

**Discussion by hipstermojo** [tailwindlabs/tailwindcss#1516] [Ideas]
I use `display: -webkit-box` when doing [text clamping](https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-line-clamp) and I can't hide that HTML element using `none` from Tailwind. I think it would be cool if `-webkit-box` could be in Tailwind so I wouldn't need this. However, `-webkit-box` is not supported in IE so I don't know if you might interested in adding it.

**Comment by royvanv** [comment]
That's the [old flexbox spec](https://css-tricks.com/old-flexbox-and-new-flexbox/), right? Can't you just use `.flex` instead?

**Comment by simonswiss** [comment]
Hey @hipstermojo 👋

This plugin that provides text clamping uitlitie susing `-webkit-box` might be interesting to you: https://www.npmjs.com/package/@neojp/tailwindcss-line-clamp-utilities

**Comment by hipstermojo** [comment]
This certainly looks like it addresses what I need. Is there a reason why this is not in the core library?

**Comment by adamwathan** [comment]
@hipstermojo Biggest reason is just only have a limited amount of resources. If I work on adding line clamp support then I'm not working on something else, like dark mode or animation or gradients or whatever. New features take a lot of time because getting the API right is very hard.

I'd like to add line-clamp eventually but until I am able to prioritize it above other projects it's best to use a plugin.

**Comment by hipstermojo** [comment]
Fair enough. I guess I'll work with what's available. I guess this issue can be closed then?

**Reply by simonswiss** [reply]
This has been moved to a discussion - so these are not effectively closed. It might become a useful response to point to when other folks have similar questions on other features 👍

**Comment by pjchender** [comment]
For those who come across this issue, note that Tailwind has already implemented the `line-clamp` class with built-in `-webkit-xxx`. If you need the line-clamp feature, please refer to the Tailwind documentation on [Line Clamp](https://tailwindcss.com/docs/line-clamp).

**Reply by Krak86** [reply]
Thank you man!

**Comment by PatelUtkarsh** [comment]
If you're hiding an element and showing it with a line clamp, it doesn't work with the following:

```
<p class="hidde sm:block line-clamp-3"></p> 
```

But since line-clamp already has `display:-webkit-box`, You can get away with the following: 
```
<p class="hidde sm:line-clamp-3"></p> 
```
